challenges/2021-05-18-sum-of-squares/solutions/Python/mrumiker.py

```diff
@@ -0,0 +1,39 @@
+import math
+
+def count_squared_addends_max_four(input_int):
+    """
+    Due to Lagrange's four square theorem all numbers are representable as the sum of at most 4 squares
+    https://en.wikipedia.org/wiki/Lagrange%27s_four-square_theorem
+
+    Solution by Corey McCarty
+    Additional input by Charles Umiker
+    """
+    upper_limit = math.floor(math.sqrt(input_int))
+    if input_int == upper_limit ** 2:
+        return 1  # 1 is the answer if and only if input_int is a perfect square
+    options = [v for v in range(1, upper_limit + 1)]
+    options.sort(reverse=True)
+    i = 0
+    answer_is_3 = False
+    while i <= len(options) - 1:
+        i_square = options[i] ** 2
+        j = i
+        while (j <= len(options) - 1):
+            j_square = options[j] ** 2
+            if input_int == i_square + j_square:
+                return 2  # Quitting early as there can't be a better solution at this point
+            k = j
+            while k <= len(options) - 1:
+                k_square = options[k] ** 2
+                if input_int == i_square + j_square + k_square:
+                    answer_is_3 = True
```

Comment:
Could always just return 3 here, right?

Reply:
since `answer_is_3` is set outside the loop, and set inside the loop, if it's ever hit it will always stay `True` so you could just return out here as @ericwburden mentioned.